walletunlocker/service_test.go

```diff
@@ -645,12 +647,20 @@ func doChangePassword(service *walletunlocker.UnlockerService, testDir string,
 		return
 	}
 
-	// Do cleanup now. Since we are in a go func, the defer at the
-	// top of the outer would not work, because it would delete
-	// the directory before we could check the content in here.
+	// Do cleanup now. Since we are in a go func, the defer at the top of
+	// the outer would not work, because it would delete the directory
+	// before we could check the content in here.
 	err = store.Close()
 	if err != nil {
 		errChan <- fmt.Errorf("could not close store: %v", err)
 		return
 	}
+
+	// The backend database isn't closed automatically if the store is
+	// closed, do that now manually.
+	err = store.Backend.Close()
+	if err != nil {
+		errChan <- fmt.Errorf("could not close db: %v", err)
```

Comment:
the linter is complaining that this error is not wrapped, however that should be a valid behavior (there are many times that we explicitly do not want to wrap an error)

Reply:
> there are many times that we explicitly do not want to wrap an error

Could you give an example? iirc the `%w` just behaves the same as `%v` with the additional wrapping property.

Also IMO it's better to wrap errors when returning an error from an interface call so you know where the call actually happens

Reply:
Yes, it's a bit annoying. But I guess it's generally good practice to wrap, so I don't mind that the linter forces us to update any code we touch. I just tend to forget about it, I should run the linter more locally before pushing a PR :)

Reply:
Yep, an example would be the errors from our DB interfaces. I would not like people relaying on `pg.ErrorNoRows` for example but the higher level that we create for that specific error `NoDataFound`.

In that case we force people to check for our error, because even if they know what underlying db is used to implement the interface, they cannot catch their specific errors.